lib/rules/no-inner-declarations.js

```diff
@@ -66,6 +76,22 @@ module.exports = {
 
     create(context) {
 
+        const sourceCode = context.sourceCode;
+        const programAst = sourceCode.ast;
+        const ecmaVersion = context.languageOptions.ecmaVersion || 6;
+        const options = context.options[1];
+        const legacy = options && options.legacy || false;
+
+        /**
+         * Check whether the linted file is in the strict mode or not.
+         * @returns {boolean} `true` if code is in strict mode.
+         */
+        function isStrictCode() {
```

Comment:
This doesn't cover all of ways to determine if code is in strict mode. It's better to check the global scope, like this:

https://github.com/eslint/eslint/blob/5aa9c499da48b2d3187270d5d8ece71ad7521f56/lib/rules/logical-assignment-operators.js#L233